docker/pravega/entrypoint.sh

```diff
@@ -102,6 +103,9 @@ configure_tier2() {
     esac
 }
 configure_segmentstore() {
+    [ ! -z "$PUBLISHED_ADDRESS" ] && add_system_property "pravegaservice.publishedIPAddress" "${PUBLISHED_ADDRESS}"
+    [ ! -z "$PUBLISHED_ADDRESS" ] && add_system_property "pravegaservice.listeningIPAddress" "0.0.0.0"
```

Comment:
I'm not clear on why you want to listen on all interfaces in the case the published address. Is this a change of behavior compared to what we have currently?

Reply:
By default, if the `listeningIPAddress` option is not set, Segmentstore will use the main interface IP address ([ref](https://github.com/pravega/pravega/blob/8ba609d3a1c497b8026f5e8739f2e6762bd02f5b/segmentstore/server/src/main/java/io/pravega/segmentstore/server/store/ServiceConfig.java#L271)). Therefore, it will only listen to requests that are sent to that IP address. However, in a scenario in which we use a different `publishedIPAddress` that is advertised to clients and used by them to connect to the Segmentstore, we want the segment store to listen on all interfaces and accept requests sent to the `publishedIPAddress` or any other internal IP that might be available. 

In a Kubernetes environment, this allows Pravega to work properly with both internal and external clients.